src/Mvc/Mvc.NewtonsoftJson/src/BsonTempDataSerializer.cs

```diff
@@ -167,13 +166,13 @@ public void EnsureObjectCanBeSerialized(object item)
 
         public override bool CanSerializeType(Type type) => CanSerializeType(type, out _);
 
-        private static bool CanSerializeType(Type typeToSerialize, out string errorMessage)
+        private static bool CanSerializeType(Type typeToSerialize, out string? errorMessage)
```

Comment:
NotNullWhen(true) on errorMessage?

Reply:
Unfortunately not. It does not produce an error if the type is unsupported.